src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -387,15 +407,22 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
             builder.cleanup();
             long termsOffset = indexOutput.getFilePointer();
 
-            OnDiskGraphIndex.write(new RemappingOnDiskGraphIndex<>(builder.getGraph(), ordinalMapper, reverseOrdinalMapper),
-                                   new RemappingRamAwareVectorValues(vectorValues, reverseOrdinalMapper),
-                                   indexOutput.asSequentialWriter());
+            var start = System.nanoTime();
+            var finalOrdinalMap = ordinalMap == null ? OnDiskGraphIndexWriter.sequentialRenumbering(builder.getGraph()) : ordinalMap;
+            try (var writer = new OnDiskGraphIndexWriter.Builder(builder.getGraph(), indexOutput, finalOrdinalMap)
+                              .with(new InlineVectors(vectorValues.dimension()))
+                              .build())
```

Comment:
Is this essentially just a migration of the index writing logic into jvector? Is there anything interesting changing by moving in this direction?

Reply:
the renumbering logic, yes, because jvector uses that when clearing deleted nodes from the graph (astra does not use jvector deletions and instead does compaction things)